### DISCUSSION TITLE
Endpoint to tokenize search query?

### DISCUSSION THREAD

**Discussion by underflow00** [meilisearch/product#794] [Feedback & Feature Proposal]
It would be useful to have an API endpoint that takes a search query string and returns a list of tokens.
A use case is to count the number of tokens the user has entered in the search box, to inform them if it has exceeded the 10 token limit.

And the list of tokens could also be returned by the Search endpoint.

**Comment by underflow00** [comment]
Somewhat relatedly, it would also be useful to have a endpoint or a way to get the "normalized" value that is used by Meilisearch for sorting. Normalized means converted to lower case, as "Uppercase letters are sorted as if they were lowercase", and also other normalization not documented.

**Comment by ManyTheFish** [comment]
Hello @underflow00,
This request is interesting; I would suggest adding a debug field in the search response that returns an array of strings containing the tokenized version of the query; what do you think? 🤔
